### DISCUSSION TITLE
A "lost" 80s home computer

### DISCUSSION THREAD

**Discussion by uzudil** [alecthomas/participle#128] [Show and tell]
I used this project for my "lost" 80s home computer: https://github.com/uzudil/benji4000
The language is vaguely like modern javascript: https://github.com/uzudil/benji4000/wiki/LanguageFeatures with built-in features to provide graphics and sound.

The benji project comes with a few games, but I also made a stand-alone retro rpg that runs on the benji4000. It's written in "bscript" which is parsed/lexed by participle: https://github.com/uzudil/svaltfen

Thanks for the great work!

**Comment by alecthomas** [comment]
That's really cool! It brings back C64 vibes. Ah the memories :)

I also like that you implemented AirWolf old school awesomeness.

**Comment by alecthomas** [comment]
This could be a great teaching language for kids.

**Comment by alecthomas** [comment]
Also have you considered writing an LSP for it? There's a pretty straightforward example [here](https://github.com/openllb/hlb/tree/master/langserver), which also uses Participle.

**Comment by uzudil** [comment]
I haven't but I will check it out, thanks!

--Gabor
